src/sdk/namespace_fs.js

```diff
@@ -2434,15 +2458,29 @@ class NamespaceFS {
             try {
                 upload_params = await this._start_upload(fs_context, undefined, undefined, params, 'w');
 
-                // the delete marker path built from its version info (mtime + ino)
                 const stat = await upload_params.target_file.stat(fs_context);
-                delete_marker_version_id = this._get_version_id_by_stat(stat);
+                if (this._is_versioning_enabled()) {
+                    // the delete marker path built from its version info (mtime + ino)
+                    delete_marker_version_id = this._get_version_id_by_stat(stat);
+                } else {
+                    // the delete marker file name would be with a 'null' suffix
+                    delete_marker_version_id = NULL_VERSION_ID;
+                }
                 const file_path = this._get_version_path(params.key, delete_marker_version_id);
 
-                // finish upload part
-                const fs_xattr = await this._assign_versions_to_fs_xattr(fs_context, deleted_version_info,
+                let fs_xattr;
+                if (this._is_versioning_suspended() &&
+                    (deleted_version_info?.version_id_str === NULL_VERSION_ID)) {
+                    fs_xattr = await this._assign_versions_to_fs_xattr(fs_context, undefined,
```

Comment:
if the deleted version was unique, it'll move to .versions/, and in that case we do need to update the prev version of the delete_marker as we do for versioning enabled